scripts/west_commands/runners/mdb.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2020 Synopsys.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for Synopsys Metaware Debugger(mdb).'''
+
+
+from os import path
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+class MdbBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for mdb.'''
+
+    def __init__(self, cfg, cores=1, jtag='digilent', nsim_args='', gui=True,
+                dig_device=''):
+        super(MdbBinaryRunner, self).__init__(cfg)
+        self.jtag = jtag
+        self.cores = cores
+        if nsim_args != '':
+            self.nsim_args = path.join(cfg.board_dir, 'support', nsim_args)
+        else:
+            self.nsim_args = ''
+        self.gui = gui
+        self.elf_name = cfg.elf_file
+        self.dig_device = dig_device
+        self.commander = 'mdb'
+
+    @classmethod
+    def name(cls):
+        return 'mdb'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash', 'debug'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--jtag', default='digilent',
+                            help='choose the jtag interface')
+        parser.add_argument('--cores', default=1,
+                            help='choose the cores to debug')
+        parser.add_argument('--gui', default=True,
+                            help='enable gui debug')
+        parser.add_argument('--nsim_args', default='',
+                            help='if given, arguments for nsim simulator')
+        parser.add_argument('--dig-device', default='',
+                            help='''choose the digilent device when
+                             --jtag=digilent''')
+
+    @classmethod
+    def create(cls, cfg, args):
+        return MdbBinaryRunner(
+            cfg,
+            cores=args.cores,
+            jtag=args.jtag,
+            nsim_args=args.nsim_args,
+            gui=args.gui,
+            dig_device=args.dig_device
+            )
+
+    def do_run(self, command, **kwargs):
+        self.require(self.commander)
+
+        mdb_basic_options = ['-nooptions', '-nogoifmain',
+                            '-toggle=include_local_symbols=1']
+
+        # nsim
+        if self.nsim_args != '':
+            mdb_target = ['-nsim', '@'+self.nsim_args]
+        # hardware target
+        else:
+            if self.jtag == 'digilent':
+                mdb_target = ['-digilent', self.dig_device]
+            else:
+                mdb_target = ['']
+
+        if command == 'flash':
+            mdb_run = ['-run', '-cl']
+        else:
+            if self.gui:
```

Comment:
Isn't that something for `do_debug`?
Also you may want to consider adding `-nogoifmain` option so that MDB doesn't try to execute application stopping at `main()` function as quite some interesting stuff will be fast-forwarded... and yes this happens by default (going up to `main()`) so we need to take care of that.